src/setup-beam.js

```diff
@@ -17,20 +20,55 @@ async function main() {
   const elixirSpec = core.getInput('elixir-version', { required: false })
   const gleamSpec = core.getInput('gleam-version', { required: false })
   const rebar3Spec = core.getInput('rebar3-version', { required: false })
+  const shouldMixRebar = core.getInput('install-rebar', { required: false })
+  const shouldMixHex = core.getInput('install-hex', { required: false })
+  const hexMirrors = core.getMultilineInput('hexpm-mirrors', {
+    required: false,
+  })
 
   if (otpSpec !== 'false') {
-    await installOTP(otpSpec, osVersion)
-    const elixirInstalled = await maybeInstallElixir(elixirSpec, otpSpec)
+    const otpVersion = await getOTPVersion(otpSpec, osVersion, hexMirrors)
+    const elixirVersion = await getElixirVersion(
+      elixirSpec,
+      otpVersion,
+      hexMirrors,
+    )
+    // No cache for Windows OTP, rebar3 and Gleam as they're downloaded from GitHub
```

Comment:
Why do we not cache things from GitHub?

Reply:
As the cache is on GitHub, what's the point of caching GitHub?

Reply:
The machines that host github action runners are not on the same network as github artifact hosting. The artifact hosting uses [Fastly CDN](https://www.fastly.com/) which we also use for repo.hex.pm. If there is network connectivity issues when trying to reach repo.hex.pm then github artifacts will have the same issues unless they have an extra layer of caching I don't know about.